index.js

```diff
@@ -194,16 +203,35 @@ IFrameClass.prototype._postRaw = function(request) {
   }
 };
 
-IFrameClass.prototype._listen = function(request) {
+IFrameClass.prototype._listen = function(options) {
+  var domain = options.domain
+  var targetWindow = options.window
+
   var self = this;
-  this._unlisten()
-  if (!self._isListening) {
-    this._messageHandlerFunc = function (event) {
-      self._messageHandler(event.originalEvent.data);
-    }
-    $(window).on('message', this._messageHandlerFunc);
-    self._isListening = true;
+
+  if (self._isListening) {
+    return
+  }
+
+  if (self._isListening) {
+    return
   }
+
+  $(window).on('message', function (e) {
+    var event = e.originalEvent
+
+    if (self.config.child && targetWindow !== window.opener) {
```

Comment:
```
targetWindow !== window.opener
```
is always going to be `false` incase of child. Do we need this?

Reply:
Removed.